pkg/services/rpcsrv/server.go

```diff
@@ -136,8 +136,18 @@ type (
 		sessionsLock sync.Mutex
 		sessions     map[string]*session
 
-		subsLock          sync.RWMutex
-		subscribers       map[*subscriber]bool
+		subsLock    sync.RWMutex
+		subscribers map[*subscriber]bool
+
+		// These two channels are aimed to manage Blockchain's subscriptions process,
+		// they are handled in a separate goroutine to avoid concurrent Blockchain's
+		// subscriptions API access. These channels are not buffered to avoid lazy
+		// subscription responses and preserve the order of subscription/unsubscription
+		// events.
+		subCh   chan interface{}
+		unsubCh chan interface{}
```

Comment:
...then why these channels? Any actions will be naturally ordered already (only one peer is changing counters and doing subscription/unsubscription).

Reply:
Yestarday I had some thought about it, but seems that just lock is enough, could you see the updated version?